ocrd_network/ocrd_network/processor_server.py

```diff
@@ -0,0 +1,205 @@
+from datetime import datetime
+import logging
+from os import getpid
+from subprocess import run, PIPE
+import uvicorn
+
+from fastapi import FastAPI, HTTPException, status, BackgroundTasks
+
+from ocrd_utils import getLogger, get_ocrd_tool_json
+from .database import (
+    DBProcessorJob,
+    db_update_processing_job,
+    initiate_database
+)
+from .models import (
+    PYJobInput,
+    PYJobOutput,
+    PYOcrdTool,
+    StateEnum
+)
+from .process_helpers import run_single_execution
+from .server_utils import (
+    _get_processor_job,
+    validate_and_resolve_mets_path,
+    validate_job_input
+)
+from .utils import (
+    calculate_execution_time,
+    generate_id,
+    tf_disable_interactive_logs
+)
+
+# TODO: Check this again when the logging is refactored
+tf_disable_interactive_logs()
+
+
+class ProcessorServer(FastAPI):
+    def __init__(self, mongodb_addr: str, processor_name: str = "", processor_class=None):
+        if not (processor_name or processor_class):
+            raise ValueError('Either "processor_name" or "processor_class" must be provided')
+        self.log = getLogger(__name__)
```

Comment:
Why though? Maybe core needs to change to `__name__`? It's a cleaner approach and always resolves to the correct `module_name.file_name` instead of manually naming the logger. So in that specific case resolves to `ocrd_network.processor_server`.

Reply:
Ok, you're right. We started with some conventions, trying to get our logs to look nice (with or without debug level) for core, processor modules and bashlib processors. Perhaps at least for core it's time to walk the Python path again. (Using `__name__` in most places instead of fixed strings, no more `ocrd_cli_wrap_processor` or `ocrd.process.profile`, `ocrd.dummy`.) Still, one difficulty there is to hide details of our implementation (like processor.helpers instead of just processor, or `ocrd_network.server_utils` instead of `ocrd_network.processing_worker` etc).

Reply:
> Still, one difficulty there is to hide details of our implementation

That's problematic indeed.